homeassistant/components/shelly/utils.py

```diff
@@ -189,7 +191,12 @@ def get_device_wrapper(hass: HomeAssistant, device_id: str):
 async def get_coap_context(hass):
     """Get CoAP context to be used in all Shelly devices."""
     context = aioshelly.COAP()
-    await context.initialize()
+    if DOMAIN in hass.data:
+        port = hass.data[DOMAIN][CONF_COAP_PORT]
+    else:
+        port = DEFAULT_COAP_PORT
```

Comment:
```suggestion
    port = hass.data[DOMAIN].get(CONF_COAP_PORT, DEFAULT_COAP_PORT)
```

Reply:
`coap_port` is the only (optional) key in DOMAIN

Reply:
you can still put `shelly: {}` in your config.

Reply:
New commit